cmd/postgres-operator/main.go

```diff
@@ -101,20 +102,27 @@ func addControllersToManager(ctx context.Context, mgr manager.Manager) error {
 func isOpenshift(ctx context.Context, cfg *rest.Config) bool {
 	log := logging.FromContext(ctx)
 
-	const openShiftAPIGroupSuffix = ".openshift.io"
+	const sccGroup, sccKind = "security.openshift.io", "SecurityContextConstraints"
 
 	client, err := discovery.NewDiscoveryClientForConfig(cfg)
 	assertNoError(err)
 
-	groups, _, err := client.ServerGroupsAndResources()
+	resourceLists, err := client.ServerResources()
 	assertNoError(err)
 
-	// If we detect that any API group name ends with "openshift.io", we'll
-	// return that this is an OpenShift environment
-	for _, g := range groups {
-		if strings.HasSuffix(g.Name, openShiftAPIGroupSuffix) {
-			log.Info("detected OpenShift environment")
-			return true
+	// If we detect that the "SecurityContextConstraints" Kind is present in the
+	// "security.openshift.io" Group, we'll return that this is an OpenShift
+	// environment
+	for _, rl := range resourceLists {
+		gv, err := schema.ParseGroupVersion(rl.GroupVersion)
+		assertNoError(err)
```

Comment:
🤔 I wonder if we should avoid parsing and errors here. Would `strings.HasPrefix(rl.GroupVersion, sccGroup+"/")` be a reasonable substitute?

Reply:
I actually went back and forth on this, and actually had `strings.HasPrefix()` initially! ... However, ultimately this felt like the safest way to ensure the proper Group was extracted from a GV, so I stuck with that approach.

Are you anticipating any issues with the current approach? (I do feel like an error would appropriate if we discover a GV that we can't parse :thinking:)

Reply:
No issues. I expect it is "impossible" for the API to return a GV that can't be parsed.

"Impossible" from the API and "impossible" in our code are different, though. I would prefer that auto-detection never halt the server because users do have a way forward when it is broken. It's not a blocker, though.

Reply:
Heh, I went back and forth again, and I def see what you mean!

While I too have no idea why an invalid GV would ever be in there in the first place, I think I agree - why let that stop PGO from running even if there were one.  We have a very high-degree of confidence those we do rely on (e.g. from the native Kube  and OpenShift API's) will be right (including the specific one we're looking for here), so why worry about errors and let others get in the way.

Updated to use `strings.HasPrefix()`.